### ISSUE TITLE
Exclude paths are not always working

### ISSUE THREAD

**Issue by sergeys-opera** [detekt/detekt#8144]
## Expected Behavior
No issues reported for the code from the excluded folders.

## Observed Behavior
Some code (but not all!) from the excluded folders is analyzed and detekt reports some issues.

## Steps to Reproduce
Here's the code I use to configure `detektAll` task in a convention plugin:
```kotlin
internal fun Project.configureDetekt() {
    val libs = libs()

    extensions.getByType(DetektExtension::class).apply {
        // Docs: https://detekt.github.io/detekt/gradle.html#options-for-detekt-configuration-closure
        buildUponDefaultConfig = true
        parallel = true
        autoCorrect = boolProperty("detekt.autoCorrect")
        toolVersion = libs.versionStr("detekt")
        config.setFrom(files("$rootDir/config/detekt/detekt.yml"))
        baseline = file("$rootDir/config/detekt/baseline.xml")
    }

    val detektAll = tasks.register<Task>("detektAll")
    tasks.named("check").dependsOn(detektAll)
    tasks.withType(Detekt::class).configureEach {
        reports {
            html {
                required.set(true)
                outputLocation.set(file("build/reports/detekt.html"))
            }
        }
        exclude("**/generated/**")
        detektAll.get().dependsOn(this)
    }

    dependencies.apply {
        add("detektPlugins", libs.library("detekt.formatting"))
        add("detektPlugins", libs.library("detekt.twitter.rules"))
    }
}

```

The main thing here is that I exclude the `**/generated/**` folder for each `Detekt` task.
I tried excluding the build folder but that didn't work well.
Running `./gradlew check` or `./gradlew detektAll` triggers all the detekt tasks (for all the build variants) in the project.
However, despite the exclude detekt runs on the generated code and I get the following code style violations:
```
/Users/serso/Projects/opera/mobilemissions/clientsdk/lib/core/build/generated/ksp/metadata/commonMain/kotlin/com/opera/momi/clientsdk/_ApiImpl.kt:11:1: Unused import [NoUnusedImports]
/Users/serso/Projects/opera/mobilemissions/clientsdk/lib/core/build/generated/ksp/metadata/commonMain/kotlin/com/opera/momi/clientsdk/_ApiImpl.kt:12:45: Unused import [NoUnusedImports]
/Users/serso/Projects/opera/mobilemissions/clientsdk/lib/core/build/generated/ksp/metadata/commonMain/kotlin/com/opera/momi/clientsdk/_ApiImpl.kt:13:23: Unused import [NoUnusedImports]
/Users/serso/Projects/opera/mobilemissions/clientsdk/lib/core/build/generated/ksp/metadata/commonMain/kotlin/com/opera/momi/clientsdk/_ApiImpl.kt:13:23: Unused import [NoUnusedImports]
``` 

## Context
I'm trying to configure detekt to run on all source code sets (android, iOS, etc) excluding the build folder.

## Your Environment

* Version of detekt used: 1.23.8
* Version of Gradle used (if applicable): 8.14
* Gradle scan link (add `--scan` option when running the gradle task):
* Operating System and version: MacOS 15.5
* Link to your project (if it's a public repository):

**Comment by Nikita-Guchakov** [comment]
Same, though I have no idea why it stoped to work, I was refactoring our gradle project. 
Don't know if it matters, but we have custom layout at one of the subprojects, so we have to use `source` param as well

fails on generated folders checks:
```kotlin
tasks.withType<Detekt>().configureEach {
   val srcDirs: Provider<List<File>> = project.getSources()
   source(srcDirs)
   exclude("**/generated/**")
}
```

as WA I tried to add:
```kotlin
tasks.withType<Detekt>().configureEach {
   source(
	project.files(srcDirs).asFileTree.matching {
		exclude("**/generated/**")
	}
   )
}
```

and it's not works either, though simple filtering works:

```kotlin
source(
   project.files(srcDirs).asFileTree.fitler{ !it.absoulutePath.contains("/generated/") } 
)
```

don't know if my understanding of gradle is not enough here, but as for plugin exclusion it's quite unexpected for me

**Comment by sergeys-opera** [comment]
Apparently this is how Gradle's source task works. The question has been answered already here: https://github.com/detekt/detekt/discussions/7589
Here seems to be an explantation from the Gradle devs: https://github.com/gradle/gradle/issues/3994
Though I don't really understand it 🤔

**Comment by sergeys-opera** [comment]
A workaround could be to write something like this:
```kotlin
exclude { it.file.path.contains("build/generated") }
```
